cmake/sip_helper.cmake

```
@@ -92,6 +92,10 @@ function(build_sip_binding PROJECT_NAME SIP_FILE)
 
     if(WIN32)
       set(MAKE_EXECUTABLE NMake.exe)
+    elseif(CMAKE_GENERATOR MATCHES "Ninja")
+      find_program(MAKE_PROGRAM NAMES make)
+      message(STATUS "Using Ninja generator with SIP requires make, found: ${MAKE_PROGRAM}")
+      set(MAKE_EXECUTABLE ${MAKE_PROGRAM})
     else()
       set(MAKE_EXECUTABLE "\$(MAKE)")
     endif()
```

Do we actually have to make this conditional on `Ninja`?  That is, on all platforms that are not Windows it seems like we should be able to just `find_program(make)` and it would do the right thing.  Like this (untested):

```suggestion
    if(WIN32)
      set(MAKE_EXECUTABLE NMake.exe)
    else()
      find_program(MAKE_PROGRAM NAMES make)
      message(STATUS "Found required make: ${MAKE_PROGRAM}")
      set(MAKE_EXECUTABLE ${MAKE_PROGRAM})
    endif()
```

---

We probably don't have to make it conditional, I was trying to make the scope as small as I know that we needed.

I think your approach is likely safer if there is some other (unknown to me) CMake generator that happens to not use make.